autogen/coding/local_commandline_code_executor.py

```diff
@@ -242,7 +262,22 @@ def _execute_code_dont_check_setup(self, code_blocks: List[CodeBlock]) -> Comman
                 f.write(code)
             file_names.append(written_file)
 
-            program = sys.executable if lang.startswith("python") else _cmd(lang)
+            # Check if the code will be executed.
+            program = None
+            if lang.startswith("python"):
+                program = sys.executable
+            else:
+                try:
+                    program = _cmd(lang)
+                except NotImplementedError:
```

Comment:
This is confusing - some languages silently get saved and ignored and some don't? Ignoring `NotImplementedError` doesn't feel right

`save_code_only` feels way too coarse to me. I would rather see a mapping of language to whether it should be executed or just saved.

Reply:
Agreed. A mapping to specify whether a given language should be executed is more suitable here. In that case we can allow user to specify the executing program as well, bringing more transparency.